lib/strategy.js

```diff
@@ -129,10 +131,22 @@ Strategy.prototype.userProfile = function(accessToken, done) {
 
       var json = JSON.parse(body);
 
-      for (var index in json) {
-        if (json[index].primary) {
-          profile.emails = [{ value: json[index].email }];
-          break;
+      if (!json || !json.length) {
+        return done(new Error('Failed to fetch user emails'));
+      }
+
+      if (self._allRawEmails) {
+        profile.emails = json.map(function (email) {
+          var mappedEmail = Object.assign(email, { value: email.email });
```

Comment:
How do you know which email is the primary one? Should we add type to these? See: http://www.passportjs.org/docs/profile/

Reply:
here's a sample `json` object:
```
[ { primary: true,
    verified: true,
    visibility: 'private',
    value: 'bob.pineapple@gmail.com' },
  { primary: false,
    verified: true,
    visibility: null,
    value: 'pineapple@users.noreply.github.com' },
  { primary: false,
    verified: true,
    visibility: null,
    value: 'bob@pineapple.com' } ]
```

Reply:
Ah, is this what the response looks like? ```[
  {
    "email": "octocat@github.com",
    "verified": true,
    "primary": true,
    "visibility": "public"
  }
]```

Reply:
Cool, sounds good.

Reply:
yep, exactly

Reply:
my `map` just transforms these objects to have a `value` key, which seems to be a convention across passport addons

Reply:
Should we add a test then?

Reply:
added tests!